starlight-rabbitmq/src/main/java/com/datastax/oss/starlight/rabbitmq/GatewayService.java

```diff
@@ -487,4 +490,27 @@ public ScheduledExecutorService getExecutor() {
   public ExecutorService getInternalExecutorService() {
     return internalExecutorService.getExecutor();
   }
+
+  public Producer<byte[]> getOrCreateProducer(String topic) {
+    return producers.computeIfAbsent(topic, this::createProducer);
+  }
+
+  public void removeProducer(String topic) {
+    Producer<byte[]> producer = producers.remove(topic);
```

Comment:
I would use compute() and ensure that we are removing a exact instance of the Producer

Reply:
Can you give more detail ? We want to remove the producer from the cache, not compute a value for it.

Reply:
done. PTAL